bindings/c/ThreadCleanup.cpp

```diff
@@ -36,25 +36,10 @@ BOOL WINAPI DllMain( HINSTANCE dll, DWORD reason, LPVOID reserved ) {
 
 static pthread_key_t threadDestructorKey;
 
-static void threadDestructor(void*) {
-	releaseAllThreadMagazines();
-}
-
 void registerThread() {
 	pthread_setspecific( threadDestructorKey, (const void*)1 );
 }
 
-static int initThreadDestructorKey() {
-	if (!pthread_key_create(&threadDestructorKey, &threadDestructor)) {
-		registerThread();
-		setFastAllocatorThreadInitFunction( &registerThread );
-	}
-
-	return 0;
-}
-
-static int threadDestructorKeyInit = initThreadDestructorKey();
```

Comment:
According to ICC this is dead code - but after taking a bit of a closer look I think you're right. I will add this code again and disable the compiler warning in this file.

Reply:
Does that mean there's a risk ICC will optimize this out? If so, we may ultimately need to devise a different way to do this that doesn't have that problem.

Reply:
I am not too familiar with the standard in this regard. But my position would be that if a compiler is allowed to optimize this out, we should fix it (independent of what icc does). And I think a compiler is allowed to do that. It would also be interesting to know what MSVC does here - because there we miss the warning for sure if there is one.